pkg/kubelet/kubelet.go

```diff
@@ -1045,7 +1051,6 @@ func (kl *Kubelet) syncPod(pod *api.BoundPod, dockerContainers dockertools.Docke
 			}
 		}
 	}
-	containersToKeep[podInfraContainerID] = empty{}
```

Comment:
@vmarmol I think the `containersInPod.RemoveContainerWithID(podInfraContainerID)` bellow has taken care of this. That one is outside the supercalifragilisticexpialidocious for-loop. Even we don't change the container, we will hit that sentence.

After this, I will try to work on the prober inside the for-loop next to make it cleaner. :)